### PR TITLE
Skip CircleCI

### PR DESCRIPTION
**Pull request by jakirkham**
This PR disables CircleCI builds when they are not needed. It does this by using CircleCI's ignore branch feature in the `circle.yml` as described by their [docs]( https://circleci.com/docs/configuration/#branches ). By creating a generic regex expression that matches any branch name, it is guaranteed to match the current branch. However, as noted in the docs, each branch could have a different `circle.yml`. So, just because one notes all branches should be skipped, another branch could have changes to `circle.yml` that allow it to run. It should also be added that this will remove `run_docker_build.sh` if Linux builds are skipped.

To test this out, we tried a couple of examples. First `c99-to-c89` is a Windows only package that is a C cross compiler converting C99 code to C89 code. It is currently not built from source so has a fast build time. We re-rendered with the changes here, the introduction of `conda-forge-build-setup` via PR ( https://github.com/conda-forge/conda-smithy/pull/196 ), the removal of `.travis.yml` for OS X skips via PR ( https://github.com/conda-forge/conda-smithy/pull/207 ), and to top it all of we used PR ( https://github.com/conda-forge/conda-smithy/pull/206 ) to disable Travis CI builds when `.travis.yml` is not present. The results of all of this can be seen in PR ( https://github.com/conda-forge/c99-to-c89-feedstock/pull/3 ).

Next one should verify that this change amounts to a no-op if Linux is not skipped. As the introduction of `conda-forge-build-setup` via PR ( https://github.com/conda-forge/conda-smithy/pull/196 ) had been tested on `cartopy` in PR ( https://github.com/conda-forge/cartopy-feedstock/pull/11 ), it made sense to also test their so as to be able to subtract out the `conda-forge-build-setup` change. As a result one can see that if there is no skip for Linux it results in PR ( https://github.com/conda-forge/cartopy-feedstock/pull/13 ). Note the only changes in there are from `conda-forge-build-setup`.

In short, this works and is well tested. IMHO this is ready to go. The only real question in my mind is whether `render_run_docker_build` and `render_circle` should be folded into the same function or not. Currently they are two separate functions, but maybe they should be folded into `render_circle`. OTOH each rerender function acts on a particular file so `render_circle` is no different in this way. Though this could be easily addressed in a subsequent PR.

### CODE REVIEW

**Comment by jakirkham** [comment]
Weird. Travis CI didn't seem to pick this up. Though there don't seem to be any failures for the webhook AFAICT. Going to close and reopen to see if that does the trick.

**Comment by jakirkham** [comment]
There we go. Now Travis CI has started.

**Comment by jakirkham** [comment]
Any thoughts on this one, @pelson?

**Comment by pelson** [comment]
> The only real question in my mind is whether render_run_docker_build and render_circle should be folded into the same function or not. 

Agreed. Let's not hold off on that though.

**Comment by pelson** [comment]
Thanks for doing this @jakirkham. Nice addition!

**Comment by jakirkham** [comment]
Of course. 😄
